internal/pkg/utils/testhelper/terminal/terminal_linux.go

```diff
@@ -0,0 +1,240 @@
+	opts = append(
+		opts,
+		expect.WithStdout(debugStdout),
+		expect.WithCloser(debugStdout),
+		expect.WithSendObserver(sendObserver(t, debugStdout)),
+		expect.WithExpectObserver(expectObserver(t, os.Stderr)),
+		expect.WithDefaultTimeout(expectTimeout),
+	)
+
+	out.Console, out.state, err = vt10x.NewVT10XConsole(opts...)
+	out.tty = &tty{file: out.Console.Tty(), closed: make(chan struct{})}
+	return out, err
+}
+
+func (c *console) Tty() Tty {
+	return c.tty
+}
+
+func (c *console) Send(s string) (int, error) {
+	c.waitBeforeSend()
+	return c.Console.Send(s)
+}
+
+func (c *console) SendLine(s string) (int, error) {
+	c.waitBeforeSend()
+	return c.Console.SendLine(s)
+}
+
+func (c *console) SendEnter() error {
+	_, err := c.Send("\n")
+	return err
+}
+
+func (c *console) SendSpace() error {
+	_, err := c.Send(" ")
+	return err
+}
+func (c *console) SendBackspace() error {
+	_, err := c.Send("\u0008")
+	return err
+}
+
+func (c *console) SendUpArrow() error {
+	_, err := c.Send("\u001B[A")
+	return err
+}
+
+func (c *console) SendDownArrow() error {
+	_, err := c.Send("\u001B[B")
+	return err
+}
+
+func (c *console) SendRightArrow() error {
+	_, err := c.Send("\u001B[C")
+	return err
+}
+
+func (c *console) SendLeftArrow() error {
+	_, err := c.Send("\u001B[D")
+	return err
+}
+
+func (c *console) ExpectEOF() (out string, err error) {
+	defer func() {
+		// Close STDIN on error (e.g. timeout)
+		if err != nil {
+			_ = c.Tty().Close()
+		}
+	}()
+
+	// Better error message
+	if str, err := c.Console.ExpectEOF(); err != nil {
+		return "", fmt.Errorf("error while waiting for EOF: %w", err)
+	} else {
+		return str, nil
+	}
+}
+
+// waitBeforeSend delays sending input.
+// We often Expect only the part of the output that is written to the console.
+// But we can send the next input only when the application expects it,
+// so all application output has to be written before.
+// The simplest solution is to wait a bit.
+func (c *console) waitBeforeSend() {
+	time.Sleep(sendDelay)
+}
+
+func (t *tty) Read(p []byte) (n int, err error) {
+	// Within the tests, interactive/Prompt.Editor is called.
+	// It starts a new OS process for the editor (to edit a longer value).
+	// In the tests, instead of the editor (vi, nano, ...), the command "true" is started, which ends immediately.
+	//
+	// The os/exec package works differently with stdin, depending on whether it is *os.File or another io.Reader.
+	// See: https://github.com/golang/go/blob/d7df872267f9071e678732f9469824d629cac595/src/os/exec/exec.go#L350-L357
+	// Our *tty implementation implements io.Reader, it is not an instance of *os.File.
+	//
+	// In os/exec, io.Copy function is called for io.Reader to redirect reader -> pipe.
+	// This blocks the test by unexpected reading from stdin, which does not happen during real execution (*os.File).
+	// For this reason, this call is terminated immediately.
+	if calledFromOsExecCmdStart() {
+		return n, io.EOF
+	}
```

Comment:
Tak toto mi zabralo viac hodin, ako je riadkov kodu :laughing: 

Nevedel som prist na to, preco mi prestal fungovat test, ked som zacal pouzivat `*tty` implementaciu.
Neskor som zistil, ze sa to deje iba v testoch kde sa pouziva `prompt.Editor()`.

Problem je v tom, ze `os/exec` ma inu implementaciu pre `*os.File`, ktore tam bolo povodne, a inu pre ostatne `io.Reader`:
https://github.com/golang/go/blob/d7df872267f9071e678732f9469824d629cac595/src/os/exec/exec.go#L350-L357

Tj. subor sa da pouzit priamo, ale pre ostatne `io.Reader` sa vytvara `pipe`, do ktorej sa kopiruju data:
https://github.com/golang/go/blob/d7df872267f9071e678732f9469824d629cac595/src/os/exec/exec.go#L362

No a tam sa test zablokoval, co sa pri realnom spusteni v OS nestane, podrobnejsie vysvetlene v komentaroch ^^^.

Reply:
Ak by som sa na to vykaslal, tak zase neviem poriesit `timeouts`, ktore uz vdaka `*tty` nenastavaju (ak nastane v teste chyba).